modin/pandas/dataframe.py

```diff
@@ -1163,7 +1163,7 @@ def join(
             if other.name is None:
                 raise ValueError("Other Series must have a name")
             other = self.__constructor__(other)
-        if on is not None:
+        if on is not None or how == "cross":
```

Comment:
After this branch there is an implementation using `concat`, but parameter `how=='concat'` is not exist for it (only for `join` and `merge`).